### ISSUE TITLE
Error installing font-lock+ on startup

### ISSUE THREAD

**Issue by mavant** [syl20bnr/spacemacs#15001]
#### Description :octocat:

font-lock+ has become an invalid package as of upstream emacs commit
4e6f98cd505ed56e7928fcdb5ab88c09d0e74d89 (August 9th 2021), or perhaps
5809728bc502d58f4fe96e98b472c569da3d8879 (August 12th).

The underlying cause is that the new function `lm-maintainers` is not quite as
permissive as the old `lm-maintainer` was; in particular it fails to parse this line:
```
;; Maintainer: Drew Adams (concat "drew.adams" "@" "oracle" ".com")
```

To resolve this it would be sufficient to do any of the following:
1. Drop the dependency from `spacemacs-modeline`. This seems like it might be fine?  AFAICT nothing in spacemacs is actually using `font-lock-ignore`.
2. Either upstream or before installing, delete or replace that line of `font-lock+.el`. I can confirm that doing this locally allows the package installation to succeed.
3. Amend or override `package-buffer-info` to tolerate failures in extracting some fields, or to understand elisp expressions like `concat`.

(although of course approaches 2 and 3 are really not spacemacs's problems per se).

#### Reproduction guide :beetle:
- Start Emacs version 28-devel at any commit > 0509f3921b52feb6b1916e4ba64c91a603991bfe
- wait

*Observed behaviour:* :eyes: :broken_heart:
Spacemacs logs an error to messages buffer on startup:
```
(Spacemacs) --> installing package: font-lock+@spacemacs-modeline... [1/2]
Fetcher: github
Source: emacsmirror/font-lock-plus
Updating /home/mavant/.emacs.d/.cache/quelpa/build/font-lock+/
Skipping rebuild of /home/mavant/.emacs.d/.cache/quelpa/packages/font-lock+-20180101.1439.el
Error getting PACKAGE-DESC: (wrong-type-argument stringp nil)
(Spacemacs) Error:
An error occurred while installing font-lock+ (error: (wrong-type-argument package-desc nil))
```

*Expected behaviour:* :heart: :smile:
Spacemacs starts without errors.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 28.0.50
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 7524e1622)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(python
 (sql :variables sql-capitalize-keywords t)
 (ocaml :variables ocaml-format-on-save t)
 helpful
 (pocket)
 csv auto-completion ivy emacs-lisp markdown org git
 (version-control :variables version-control-diff-tool 'diff-hl)
 (shell-scripts :variables shell-scripts-format-on-save t)
 (shell :variables shell-default-height 30 shell-default-position 'bottom)
 (spell-checking :variables enable-flyspell-auto-completion t)
 treemacs multiple-cursors
 (xclipboard :variables xclipboard-enable-cliphist t))
```
- System configuration features: CAIRO DBUS FREETYPE GIF GLIB GMP GNUTLS GSETTINGS HARFBUZZ IMAGEMAGICK JPEG JSON LCMS2 LIBSELINUX LIBXML2 MODULES NATIVE_COMP NOTIFY INOTIFY PDUMPER PNG RSVG SECCOMP SOUND THREADS TIFF TOOLKIT_SCROLL_BARS X11 XDBE XIM XPM GTK3 ZLIB

#### Backtrace :paw_prints:
```

```

**Comment by lebensterben** [comment]
To be honest I don't know why this package is used.
It says it's pulled in since it's a dependency of `spaceline-all-the-icons`. But the latter doesn't mention anything about `font-lock+`, used the function `put-text-property-unless-ignore`, or used the property `font-lock-ignore`.

https://github.com/syl20bnr/spacemacs/blob/3875f7a2fdef2f7373a1f01622dc228b1ed2b534/layers/%2Bspacemacs/spacemacs-modeline/packages.el#L29-L35

We probably should remove this entirely.

**Comment by emacs18** [comment]
I am unable to reproduce the problem reported in this issue, i.e, my emacs 28 has no problem installing font-lock+ package.  I tested emacs 28 build from about a month ago as well as emacs 28 built today using latest source code.  This is on Ubuntu 20.04.

**Comment by mavant** [comment]
I can confirm it's working for me now, too:
```
*** Welcome to IELM ***  Type (describe-mode) for help.
ELISP> (lm-maintainers "font-lock+.el")
(("Drew Adams" . "drew.adams@oracle.com"))
```

I think this was fixed upstream in [22d8e71d0462be77cb1c7999b36713d0bdfd65af](https://github.com/emacs-mirror/emacs/commit/22d8e71d0462be77cb1c7999b36713d0bdfd65af) in response to [bug 50049](https://debbugs.gnu.org/db/50/50049.html)
